packages/pigeon/lib/generator_tools.dart

```diff
@@ -219,3 +219,29 @@ void addLines(Indent indent, Iterable<String> lines, {String? linePrefix}) {
     indent.writeln('$prefix$line');
   }
 }
+
+/// Recursively merges [y] into [x].  In other words, whenever there is a
+/// conflict over the value of a key path, [y] is selected.
+Map<String, Object> mergeMaps(Map<String, Object> x, Map<String, Object> y) {
```

Comment:
It would be easier to reason about calls to this method if it used two named parameters with meaningful names like "base" and "overrides" or "source" and "target", rather than having to read the comment and then remember the semantic meaning of the two arbitrary names.

Reply:
Done, I went with "source" and "modification".